src/Swashbuckle.AspNetCore.Annotations/SwaggerParameterAttributeOperationFilter.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.AspNetCore.Mvc.Controllers;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Swashbuckle.AspNetCore.Annotations
+{
+    /// <summary>
+    /// An operation filter that adds parameter metadata on an action.
+    /// defined by the <see cref="SwaggerParameterAttribute" /> on controller methods.
+    /// </summary>
+    /// <remarks>
+    /// This filter overrides any existing description and sets the Required option only if set to true, otherwise it leaves it alone.
+    /// </remarks>
+    /// <seealso cref="Swashbuckle.AspNetCore.SwaggerGen.IOperationFilter" />
+    public class SwaggerParameterAttributeOperationFilter : IOperationFilter
+    {
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            EnsureParameters(operation);
+
+            foreach (var pair in this.GetParameterAttributes(context))
+            {
+                var param = GetOrAddParameter(operation, pair.Key);
+                param.Description = pair.Value.Description;
+
+                // Only override if required is set to true, meaning that if required is not set it will be whatever the implementation has detected it should be.
+                // This allows for use of description without having to worry about the required parameter as well - though it prevents to option of using it to se a parameter as not required if the implementation thinks it is!
+                param.Required = pair.Value.Required ? true : param.Required;
+            }
+        }
+
+        /// <summary>
+        /// Ensures the parameters exists on the operation
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        private static void EnsureParameters(Operation operation)
+        {
+            if (operation.Parameters == null)
+            {
+                operation.Parameters = new List<IParameter>();
+            }
+        }
+
+        /// <summary>
+        /// Gets the or add parameter to the operation collection of parameters
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="name">The name of the parameter</param>
+        /// <returns></returns>
+        private static IParameter GetOrAddParameter(Operation operation, string name)
+        {
```

Comment:
I don't think the conditional addition is necessary here. I can't think of a scenario where you would have a request parameter in the method signature that hasn't already been mapped to a Swagger `Parameter` by the generator. Furthermore, if we were to add parameters here - they would need to have values for `Name` and `In` to be valid.

Reply:
Actually the only reason for that was the testing. It failed because when i setup the test there was nothing (the generator is not there when testing).

I could just change the test to introduce the parameter, but was uncertain if it could happen in a proper runtime scenario.